common-rest/src/main/java/org/uniprot/api/common/repository/solrstream/FacetStreamExpression.java

```diff
@@ -28,99 +28,54 @@ public enum MetricFunctionName {
     }
 
     public FacetStreamExpression(
-            String collection,
-            String facet,
-            SolrStreamFacetRequest request,
-            FacetConfig facetConfig) {
+            String collection, SolrRequest request, SolrFacetRequest facetRequest) {
         super("facet");
-        validateParams(
-                collection,
-                request.getQuery(),
-                facet,
-                request.getMetrics(),
-                request.getBucketSorts(),
-                request.getBucketSizeLimit());
 
-        FacetProperty facetProperty = facetConfig.getFacetPropertyMap().get(facet);
+        validateParams(collection, request, facetRequest.getName());
         this.addParameter(new StreamExpressionValue(collection));
-        this.addParameter(new StreamExpressionNamedParameter("q", request.getQuery()));
-        this.addParameter(new StreamExpressionNamedParameter("buckets", facet));
+        this.addParameter(new StreamExpressionNamedParameter("q", constructQuery(request)));
+        this.addParameter(new StreamExpressionNamedParameter("buckets", facetRequest.getName()));
         this.addParameter(
-                new StreamExpressionNamedParameter(
-                        "bucketSorts", getBucketSorts(request, facetProperty)));
+                new StreamExpressionNamedParameter("bucketSorts", getBucketSorts(facetRequest)));
         this.addParameter(
                 new StreamExpressionNamedParameter(
-                        "bucketSizeLimit", String.valueOf(request.getBucketSizeLimit())));
-        List<StreamExpression> metricExpressions = parseMetrics(request.getMetrics());
-        this.getParameters().addAll(metricExpressions);
+                        "bucketSizeLimit", getBucketSizeLimit(facetRequest)));
+        StreamExpression expression =
+                new StreamExpression(MetricFunctionName.count.name()).withParameter("*");
```

Comment:
Earlier we were getting list of StreamExpression for metrics. Now, we have hardcoded only one metric count(*). Do we only need one metric?

Reply:
Yes, we currently only uses it.